### ISSUE TITLE
Ensure player has plot-critical items at start of each level

### ISSUE THREAD

**Issue by Username-N00b-is-not-available** [Realm667/WolfenDoom#838]
Since we started working on coop, I thought that it might be an improvement to add checks for required mission items in players' inventory. If they want to warp directly to some map, as could happen when a player logs out and joins again in the middle of a coop session, they won't have to cheat for critical items they would anyway have had in their inventory, had they stayed in the game for all the time. Obviously this will also be useful in singleplayer games too, e.g. for testing. This will not include secret/optional items such as Eisenmann files and Mayan artifacts.

List of items that should be checked for, IMO:
- C1M6: the solar disc is already given at map start via scripts if the player has none.
- C2M5_B: 10 explosive charges, for destroying the complex.
- C3M5_B and all subsequent maps until C3M6_B: Firebrand (well, obviously).
- C3M5_B and C3M5_C: Kennkarte, to pass through the officer.

I would like to gather the full list with the team's help (I could easily have forgotten a few items), then implement it in maps myself.
What about also ensuring on _B/_C subsequent maps (where appropriate) that the player has weapons that were issued to him for _A? Please post your thoughts on this, too.

@Ozymandias81 @Tormentor667

**Comment by AFADoomer** [comment]
This would be easy enough to implement with an event handler and a data file listing required items for each map....  Or by just giving the items via ACS, like the solar disc is already handled.

**Comment by Ozymandias81** [comment]
I still have to play the whole thing, so unfortunately I am not the right person to ask about this. We must also get rid of weapons on c3m6_A am I right? Once we land from the blimpship...

**Comment by Tormentor667** [comment]
Why do the weapons on C3M6_A need to be removed?

**Comment by Username-N00b-is-not-available** [comment]
ACS would be the easiest, I intended using it for the matter. I just need to know which maps are in need of this...
@Ozymandias81 I must say I admire your abiltities for abstinence :-) As for C3M6_A @Tormentor667, in older versions of the game the player's weapons were removed from inventory at the ending of C3M5_C (see MARINEAFTERDOUGLESCONV31 ```\cQSgt. Ascher\n\cPCAPTAIN!!!\cC LEAVE YOUR WEAPONS HERE, THEY ARE TOO HEAVY FOR THE PARACHUTES!!! HURRY NOW!!!``` for proof), but the behaviour was cancelled at some point, and some months ago I suggested it to be restored, because with the player being stacked on weapons and ammo the map became way easier than it was. If the player drops the weapons, Hosenscheisser would also become a greater aid later on.

My plan on the removal: strip player off every weapon, except obviously Tyrfing/Firebrand (I don't think knife is needed anymore, does the player have it at C3M5_A?) and a sidearm (Walther, since it was given in C3M5_A) + 24 or 32 9mm bullets. Inventory items should not be touched except grenades (I think we can leave three on the player).
Here is a list of weapons that are available on C3M6_A and C3M6_B (green = present as editor-spawned pickup, red = absent/only dropped from enemies):
![image](https://user-images.githubusercontent.com/59409713/122449884-0aa8e780-cfaf-11eb-8c09-bc223b0bb7ea.png)
It means that the player will not have (the shovel,) the autoshotgun, the Nebelwerfer and the UMG43 on the final level. They could be placed somewhere en route in one of these two maps, maybe one of them in a broken crate that was launched off the airship with its explosion and landed somewhere in the station (possibly tearing through one of the houses' roofs :-) )
***
If the team considers the weapon removal undesirable, I won't insist.
